nextcord/interactions/application.py

```diff
@@ -0,0 +1,808 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+from .. import utils
+from ..embeds import Embed
+from ..enums import InteractionResponseType, InteractionType
+from ..errors import ClientException, HTTPException, InteractionResponded, InvalidArgument
+from ..file import File
+from ..flags import MessageFlags
+from ..message import Attachment
+from ..types.snowflake import Snowflake
+from ..webhook.async_ import WebhookMessage, async_context, handle_message_parameters
+from .base import (
+    Interaction,
+    InteractionMessage,
+    InteractionResponse,
+    PartialInteractionMessage,
+    _InteractionMessageState,
+)
+
+__all__ = ("ApplicationCommandInteraction", "ApplicationAutocompleteInteraction")
+
+if TYPE_CHECKING:
+    from ..application_command import (
+        BaseApplicationCommand,
+        SlashApplicationSubcommand,
+        SlashOptionData,
+    )
+    from ..message import AllowedMentions
+    from ..state import ConnectionState
+    from ..types.interactions import (
+        ApplicationAutocompleteInteraction as ApplicationAutocompletePayload,
+        ApplicationCommandInteraction as ApplicationCommandPayload,
+        ApplicationCommandInteractionData as InteractionData,
+        ApplicationCommandInteractionDataOption as OptionsPayload,
+    )
+    from ..ui.modal import Modal
+    from ..ui.view import View
+
+MISSING: Any = utils.MISSING
+
+
+class ApplicationCommandInteraction(Interaction):
+    """Represents the interaction for all application commands.
+
+    This interaction is a subclass of :class:`Interaction`.
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    Attributes
+    ----------
+    application_command: Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        The application command that triggered the interaction.
+    app_command_name: :class:`str`
+        The name of the application command that triggered the interaction.
+    app_command_id: :class:`int`
+        The application command ID that triggered the interaction.
+    options: List[SlashOptionData]
+        The application command options that have been given a value.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "application_command",
+        "app_command_name",
+        "app_command_id",
+        "options",
+    )
+
+    def __init__(self, *, data: ApplicationCommandPayload, state: ConnectionState) -> None:
+        super().__init__(data=data, state=state)
+
+        self.application_command: Optional[
+            Union[SlashApplicationSubcommand, BaseApplicationCommand]
+        ] = None
+
+    def _from_data(self, data: ApplicationCommandPayload) -> None:
+        super()._from_data(data=data)
+
+        self.data: InteractionData = data.get("data")
```

Comment:
They subclasses do in fact use the correct types. The imports follow similar patter to the following: 
`from ..types.interactions import ApplicationCommandInteractionData as InteractionData`

Reply:
Ah, I missed that part. I don't see why it should be aliased to `InteractionData`, it makes the code a little confusing to read through.

Reply:
I aliased them for the following reasons:
- It was aliased in the original Interaction file
- For the sake of reducing the length of the typehints since they are quite big
- For the sake of simplifying things a bit

Reply:
1. No it wasn't? I just looked at the source code, and only the `Interaction` typed dict was aliased to not conflict with the actual `Interaction` class defined in the file.
2. That's not a strong argument to use this alias. To me, the whole point of type hints is not only for static type checkers, but also for people reading the code to understand what a specific variable/attribute/parameter is. Shortening it like how you did it here makes it less obvious what specifically the type of the `data` attribute is. Plus, we already have linters to take care of long lines.
3. How would this alias simplify things? I don't understand.

Reply:
Regarding point 3: My thought was mainly to Simply the class name, since `ApplicationCommandInteractionData` was very long and a bit overwhelming, and I thought the alias simplified things and in result, allowed for better understanding of the code - Albeit, based on your counter arguments, I can see the flaws and reasons for why this isn't a good idea; will be changing the code